cishouseholds/validate.py

```diff
@@ -1,10 +1,41 @@
 import csv
+from datetime import datetime
+
+from cerberus import TypeDefinition
+from cerberus import Validator
+from pyspark.sql import Row
+
+from cishouseholds.pyspark_utils import ListAccumulator
 
 
 class InvalidFileError(Exception):
     pass
 
 
+class PysparkValidator(Validator):
+    types_mapping = Validator.types_mapping.copy()
+    types_mapping["timestamp"] = TypeDefinition("timestamp", (datetime,), ())
+
+
+def filter_and_accumulate_validation_errors(
+    row: Row, accumulator: ListAccumulator, cerberus_validator: Validator
+) -> bool:
+    """
+    Validate rows of data using a Cerberus validator object, filtering invalid rows out of the returned dataframe.
```

Comment:
May be my understanding, but should the Cerberus validator object be the derived class PySparkValidator?

In the example may be worth including how `validator` is created?

Reply:
Great point. Doesn't need to be our custom one, so have added a basic example and clarified what the custom `Validator` class is for.